cmd/clusterctl/phases/pivot.go

```diff
@@ -309,7 +380,7 @@ func moveMachines(from sourceClient, to targetClient, machines []*clusterv1.Mach
 	klog.V(4).Infof("Preparing to move Machines: %v", machineNames)
 
 	for _, m := range machines {
-		if m.DeletionTimestamp != nil {
+		if m.DeletionTimestamp != nil && !m.DeletionTimestamp.IsZero() {
```

Comment:
Just the `IsZero()` call is sufficient (it handles nil correctly).

Reply:
Actually this is necessary because we get all Machines regardless if they are or not in a cluster. We need to filter because the objects in process of deletion.